commons/com.b2international.index.es8/src/com/b2international/index/es8/Es8Client.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.index.es8;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.RestClientBuilder.HttpClientConfigCallback;
+import org.elasticsearch.client.RestClientBuilder.RequestConfigCallback;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.json.jackson.JacksonJsonpMapper;
+import co.elastic.clients.transport.ElasticsearchTransport;
+import co.elastic.clients.transport.rest_client.RestClientTransport;
+
+/**
+ * Special Elasticsearch 8 compatible Java Client that creates its own Java HTTP client and a {@link ElasticsearchClient} and makes it available to
+ * the index services.
+ * 
+ * @since 8.5
+ */
+public class Es8Client implements Closeable {
+
+	private final HttpHost host;
+	
+	private final ElasticsearchTransport transport;
+	private final ElasticsearchClient client;
+
+	public Es8Client(String clusterName, String clusterUrl, String username, String password, int connectTimeout, int socketTimeout, SSLContext sslContext, ObjectMapper mapper) {
+		this.host = HttpHost.create(clusterUrl);
+		
+		final RequestConfigCallback requestConfigCallback = requestConfigBuilder -> requestConfigBuilder
```

Comment:
Is the 100 MB &rarr; 1 GB request limit increase also applicable to this client somehow? What is the default here, and can it be changed via this callback or the one below?

This is the part I'm thinking of:

https://github.com/b2ihealthcare/snow-owl/blob/871a777ade4fdc85f6a209771f05b9d07b4e9c8b/commons/com.b2international.index/src/com/b2international/index/es/client/http/EsHttpClient.java#L59-L72

Reply:
Right, that needs to be applied here as well. I think it would be better to create only a single HTTP client, will take a look whether that is possible or not. The new ES8 client library uses a newer version of the Apache HTTP, hopefully we can migrate to that.

Reply:
Done